src/geo_traits_impl/mod.rs

```diff
@@ -0,0 +1,57 @@
+mod coord;
+mod geometry;
+mod geometry_collection;
+mod line_string;
+mod multi_line_string;
+mod multi_point;
+mod multi_polygon;
+mod point;
+mod polygon;
+
+// These structures are needed because we can't implement traits on types like
+// `geojson::PointType` because they are just type aliases of raw types like
+// `Vec<f64>`.
+
+#[derive(bytemuck::TransparentWrapper)]
+#[repr(transparent)]
+pub struct PointType(crate::Position);
```

Comment:
I wonder if instead we should migrate all the geojson types from type aliases to actual types - the TransparentWrapper stuff seems expedient, but not flexible.

I don't think we necessarily need to address that in order to merge this PR - the current proposal is all within this new module, so I'm not overly concerned.

Reply:
I'd be in favor of creating full types. I don't have a preference on whether that should happen in this PR or not. That would obviate the need for using bytemuck here though

Reply:
Slightly in favor of merging with the current design and iterating with dedicated types after, but I don't feel strongly about that